frontend/src/pages/ProjectSettings/GitHubSettingsModal/GitHubSettingsModal.tsx

```diff
@@ -0,0 +1,190 @@
+}
+
+export const GitHubSettingsModal = ({
+	service,
+	githubRepos,
+	handleSave,
+	githubIntegrated,
+	closeModal,
+}: Props) => {
+	const handleSubmit = (formValues: FormValues) => {
+		handleSave(service, formValues.githubRepo)
+		closeModal()
+	}
+
+	if (!service) {
+		return null
+	}
+
+	return (
+		<Modal
+			onCancel={closeModal}
+			visible={!!service}
+			title={`GitHub settings for ${service.name} service`}
+			width="600px"
+		>
+			<ModalBody>
+				{!githubIntegrated ? (
+					<Stack>
+						<Text size="small" weight="medium">
+							It looks like you have not enabled GitHub to
+							integrate with Highlight yet. Visit{' '}
+							<Link to={`/${service.projectID}/integrations`}>
+								Integrations
+							</Link>{' '}
+							to start creating GitHub issues from comments,
+							enhancing your error stacktraces with more context,
+							and more!
+						</Text>
+						<Box display="flex">
+							<LinkButton
+								to={`/${service.projectID}/integrations`}
+								trackingId="github-settings-integrations-page-link"
+							>
+								Integrate GitHub
+							</LinkButton>
+						</Box>
+					</Stack>
+				) : (
+					<GithubSettingsForm
+						service={service}
+						githubRepos={githubRepos}
+						handleSubmit={handleSubmit}
+						handleCancel={closeModal}
+					/>
+				)}
+			</ModalBody>
+		</Modal>
+	)
+}
+
+type GithubSettingsFormProps = {
+	service: Service
+	githubRepos: GitHubRepo[]
+	handleSubmit: (formValues: FormValues) => void
+	handleCancel: () => void
+}
+
+const GithubSettingsForm = ({
+	service,
+	githubRepos,
+	handleSubmit,
+	handleCancel,
+}: GithubSettingsFormProps) => {
+	const githubOptions = useMemo(
+		() =>
+			githubRepos.map((repo: GitHubRepo) => ({
+				id: repo.key,
+				label: repo.name.split('/').pop(),
+				value: repo.repo_id.replace(
+					'https://api.github.com/repos/',
+					'',
+				),
+			})),
+		[githubRepos],
+	)
+
+	const form = Form.useFormState<FormValues>({
+		defaultValues: { githubRepo: service.githubRepoPath || null },
+	})
+
+	return (
+		<Form state={form} onSubmit={() => handleSubmit(form.values)}>
+			<Stack>
+				<Text size="small">
+					Connect <i>{service.name}</i> to a GitHub repository to
+					enable enhancements when debugging errors.
+				</Text>
+				<Box display="flex" alignItems="center" gap="16">
+					<Select
+						aria-label="GitHub Repository"
+						className={styles.repoSelect}
+						placeholder="Search repos..."
+						onSelect={(repo: string) =>
+							form.setValue(form.names.githubRepo, repo)
+						}
+						value={form.values.githubRepo?.split('/').pop()}
+						options={githubOptions}
+						notFoundContent={<span>No repos found</span>}
+						optionFilterProp="label"
+						filterOption
+						showSearch
+					/>
+
+					<Button
+						kind="danger"
+						trackingId="remove-repo-from-service"
+						size="medium"
+						onClick={() =>
+							form.setValue(form.names.githubRepo, null)
+						}
+						disabled={!form.values.githubRepo}
```

Comment:
Not sure why but this button looks like it's disabled even when the repo is set. You have to hover over it to realize that you can indeed click on it.

![Kapture 2023-08-11 at 15 28 06](https://github.com/highlight/highlight/assets/58678/b99db3ad-a55a-4feb-b3b8-c383d1ded9e7)

Reply:
Yeah, I can take a look here

Reply:
This just seems to be how we designed the danger button: https://github.com/highlight/highlight/blob/main/packages/ui/src/components/Button/styles.css.ts#L117

I'm not too worried here to create new styles because going to have Julian redesign this modal before rolling out